src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionTests.cs

```diff
@@ -157,4 +162,44 @@ public async Task HttpRequestAndErrorResponseLogged()
         Assert.Equal("SendingHttpRequest", writeList[0].EventId.Name);
         Assert.Equal("UnsuccessfulHttpResponse", writeList[1].EventId.Name);
     }
+
+    [Fact]
+    public async Task NegotiateAsyncAppendsCorrectAcceptHeader()
+    {
+        var testHttpHandler = new TestHttpMessageHandler(false);
+        var negotiateUrlTcs = new TaskCompletionSource<bool>();
```

Comment:
nit: `TaskCreationOptions.RunContinuationsAsynchronously` for all the `TaskCompletionSource`s